apis/v1alpha2/httproute_types.go

```diff
@@ -664,7 +699,42 @@ type HTTPRequestHeaderFilter struct {
 	Remove []string `json:"remove,omitempty"`
 }
 
-// HTTPRequestRedirectFilter defines configuration for the RequestRedirect filter.
+// HTTPPathModifierType defines the type of path redirect.
+type HTTPPathModifierType string
+
+const (
+	// This type of modifier indicates that the complete path will be replaced
+	// by the path redirect value.
+	AbsoluteHTTPPathModifier HTTPPathModifierType = "Absolute"
+
+	// This type of modifier indicates that any prefix path matches will be
+	// replaced by the substitution value. For example, a path with a prefix
+	// match of "/foo" and a ReplacePrefixMatch substitution of "/bar" will have
+	// the "/foo" prefix replaced with "/bar" in matching requests.
```

Comment:
Is this a path-token match or a string match?  If the input path is `/foo/qux` it will become `/bar/qux`, yes?  What if the input was `/fooqux` ?

Reply:
the ingress one is `path` token https://kubernetes.io/docs/concepts/services-networking/ingress/#path-types , should this be the same?

Reply:
Good catch, this was not clearly documented in the context of rewrites and redirects, I've filed https://github.com/kubernetes-sigs/gateway-api/pull/1189 to fix that.